packages/bpk-component-bottom-sheet/src/BpkBottomSheet.tsx

```diff
@@ -15,86 +15,145 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { useState, type ReactNode } from "react";
+import type { ReactNode } from "react";
+import { useState } from "react";
 
-import { Portal, cssModules } from "../../bpk-react-utils";
-import { withScrim } from "../../bpk-scrim-utils";
+import BpkBreakpoint, { BREAKPOINTS } from "../../bpk-component-breakpoint";
+// @ts-expect-error Untyped import. See `decisions/imports-ts-suppressions.md`.
+import { BpkButtonLink } from '../../bpk-component-link';
+// @ts-expect-error Untyped import. See `decisions/imports-ts-suppressions.md`.
+import BpkCloseButton from '../../bpk-component-close-button';
+import { BpkDialogWrapper, cssModules } from "../../bpk-react-utils";
+import BpkNavigationBar from "../../bpk-component-navigation-bar";
 
 import STYLES from './BpkBottomSheet.module.scss';
-import BpkBottomSheetInner from "./BpkBottomSheetInner";
-
 
 const getClassName = cssModules(STYLES);
-const ScrimBpkBottomSheetInner = withScrim(BpkBottomSheetInner)
 
 export type Props = {
   actionText?: string;
+  ariaLabelledby: string;
   children: ReactNode;
   closeLabel?: string;
   closeOnEscPressed?: boolean;
   closeOnScrimClick?: boolean;
   id: string;
-  isOpen: boolean;
   onAction?: () => void;
   onClose: (
-    arg0?: TouchEvent | MouseEvent | KeyboardEvent,
+    arg0?: Event,
```

Comment:
Should we be using generic types here which could introduce issues with types down the line?

Reply:
Can you elaborate? Are you talking about type issues or issues with the events?

Reply:
We initially used `TouchEvent | MouseEvent | KeyboardEvent` which resulted in a type error in the outer `BpkDialogueWrapper`, as we pass it an `onClose` function that it uses to handles the backdrop click and on escape key down. 

Without a generic `Event` type, we were running into a wall with the escape event throwing a type error relating to the mismatch type use of `SyntheticEvent<HTMLDialogElement, Event>` to the `TouchEvent | MouseEvent | KeyboardEvent` type casted as above. Using the generic `Event` type allows us to satisfy the `SyntheticEvent<HTMLDialogElement, Event>` type. 

Using the `SyntheticEvent<HTMLDialogElement, Event>` type to type check these event does not satisfy type checking as the other `onClose` calls in the `BpkDialogueWrapper` are using the native `addEventListener` which does not accept a `SyntheticEvent`, only accepting a listener on `MouseEvent` + 1 overload (which in our case, we are using a `PointerEvent`).

Do you have any suggestions on how to overcome this?

Reply:
We have ended up using union events `TouchEvent | MouseEvent | KeyboardEvent | SyntheticEvent<HTMLDialogElement, Event>`